client/cgo/verification.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import "C"
+
+import (
+	"bytes"
+	"unsafe"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+// main is required to build a shared library, but does nothing
+func main() {}
+
+//export cgoVerifySignature
+func cgoVerifySignature(pk unsafe.Pointer, pkSize C.int,
+	message unsafe.Pointer, size C.int,
+	sig unsafe.Pointer, sigSize C.int) C.int {
+	if int(pkSize) < sign.PublicKeySize {
+		return 0
+	}
+	pkBytes := C.GoBytes(pk, pkSize)
+	messageBytes := C.GoBytes(message, size)
+	sigBytes := C.GoBytes(sig, sigSize)
+	key := sign.PublicKey(pkBytes)
+	if key.Verify(messageBytes, sigBytes) {
+		return 1
+	}
+	return 0
+}
+
+//export cgoVerifyVrf
+func cgoVerifyVrf(pk unsafe.Pointer, pkSize C.int,
+	m unsafe.Pointer, size C.int,
+	index unsafe.Pointer, indexSize C.int,
+	proof unsafe.Pointer, proofSize C.int) C.int {
+	if int(pkSize) < vrf.PublicKeySize ||
+		int(indexSize) < vrf.Size ||
+		int(proofSize) < vrf.ProofSize {
+		return 0
+	}
+	pkBytes := C.GoBytes(pk, pkSize)
+	mBytes := C.GoBytes(m, size)
+	vrfBytes := C.GoBytes(index, indexSize)
+	proofBytes := C.GoBytes(proof, proofSize)
+	var key vrf.PublicKey
+	copy(key[:], pkBytes)
+	if key.Verify(mBytes, vrfBytes, proofBytes) {
+		return 1
+	}
+	return 0
+}
+
+//export cgoVerifyHashChain
+func cgoVerifyHashChain(prevHash unsafe.Pointer, root unsafe.Pointer, size C.int) C.int {
+	if int(size) != crypto.HashSizeByte {
+		return 0
+	}
+	prevHashBytes := C.GoBytes(prevHash, size)
+	rootBytes := C.GoBytes(root, size)
+	if bytes.Equal(prevHashBytes, rootBytes) {
```

Comment:
See above (https://github.com/coniks-sys/coniks-go/pull/39/files#r73228663), but shouldn't `rootBytes` be hashed?